kubeflow/pipeline/pipeline-ui.libsonnet

```diff
@@ -185,5 +185,34 @@
         },
       },
     },  // deployUi
+
+    crd: {
```

Comment:
You don't need to add the crd definition here. Instead, you want to give ml-pipeline-ui service account CRUD permission on the `viewer` resource. To do this, modify the service account role definition above, which currently allows `ml-pipeline-ui` permissions to modify pods and pods/logs. Add the following rules as well:

```
 {
          apiGroups: [
            "kubeflow.org",
          ],
          resources: [
            "viewers",
          ],
          verbs: [
            "create",
            "get",
            "list",
            "watch",
            "update",
            "patch",
            "delete",
          ],
        },
```

These are the rules the viewer controller has for modifying `viewer` resources here:
https://github.com/kubeflow/kubeflow/blob/4b1ae00a997cd83b114c5b00d53a0928cdbcb81c/kubeflow/pipeline/pipeline-viewercrd.libsonnet#L54

Reply:
Done. Thank you!